Add support for multiple ignored directives

skaldo opened the pull request:
Sometimes it is useful to have a possibility to ignore more than one directive.

skaldo committed:
Add support for multiple ignored directives

Sometimes it is useful to have a possibility to ignore more than one directive.

Narretz commented:
Hi, I have a few questions. What exactly does this change and why do we need it? Also, tests are failing and the change itself must be tested.

skaldo commented:
Hi, this change allows us to pass an array containing the names of the directives, we don't want to compile for some reason, to the $compile function.
Regarding tests, I've run them on my fork and the last two have passed, the 1st and 2nd failed because of timeouts..

Narretz commented:
Travis says there are some failures in the e2e tests. They don't look like they are timeouts / flakes. Also, we still need a test for your change.

skaldo committed:
test

test

skaldo committed:
Update compile.js

try removal of the additional arguments.

gkalpak commented on src/ng/compile.js:
```diff
@@ -2104,6 +2104,11 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
+        for (var i = 0; i < ignoreDirective.length; i++) {
+          if (name === ignoreDirective[i]) return null;
+        }
+      }
```
This whole block can be reduced to:

```js
if (isArray(ignoreDirective) && includes(ignoreDirective, name)) return null;
```

gkalpak commented:
I am a bit sceptical, because this is undocumented (private) API, but this change will not be serving the needs of Angular core.

skaldo commented:
Well, in our application this change is really beneficial. On the other hand, if it's going to be useless for the core and if it's big job to make it working, let's close this request.

Anyway, what's failing the tests is the recursive call of compileNodes function with the ignoreDirective parameter.

caitp commented:
you know, this probably could be useful for internal purposes, like helping people build recursive tree structured directives easily using the directive API. It would need a bit more than that, but I can see how you could build that kind of API on top of it.

skaldo committed:
Fix of the failing tests.

The change now conforms the tests. The @gkalpak's simplification is
included. Thanks!

skaldo committed:
Tests for ignoring directives

I have added tests for both one ignored directive (string parameter) and
the multiple ignored directives (array parameter).

skaldo committed:
Fix of the test fail in IE and FF

gkalpak commented:
As pointed out by @Narretz, this is essentially the same functionality as #14635. Although this also has test coverage, #14635's approach is more powerful/flexible, so I'm going to close this in favor of #14635.
<sub>(I doubt we are likely to implement this feature, but if we would #14635 would be the way to go.)</sub>
